be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -80,9 +129,17 @@ class SinkBuffer {
         }
     }
 
-    bool is_full() const { return _pending_chunks.get_size() >= config::pipeline_io_cache_size; }
+    bool is_full() const {
```

Comment:
is_full is unsafe because of absence of lock guard.  when the number of ExchangeSinkOperator instance is large, the queue is 1 less than its' capacity, this queue would overflow and be extended, the the peeking operation on the head of the queue is unsafe.

Reply:
The container of queue is list not vector, it has no no scaling operation, so it's safe that just read the front of the list